harness/determined/keras/_tf_keras_trial.py

```diff
@@ -802,10 +802,20 @@ def _allreduce_logs(self, logs: Dict) -> Dict:
         logging.debug(f"all-reducing logs on worker {hvd.rank()} for {len(keys)} keys {keys}.")
         return {key: np.array(hvd.allreduce(logs[key], name=key)) for key in keys}
 
+    def _convertPossibleTensor(self, possibleTensor: Any) -> Any:
+        if isinstance(possibleTensor, EagerTensor):
+            # Horovod will promote an int to a tensor in eager mode.
```

Comment:
probably worth re-writing this comment to be up-to-date, as it seems horovod is no longer the only source of this problem, right?

Reply:
Yeah I wasn't sure if it was Horovod, but then some other change in Tensor triggered it for other things. I wasn't able to find anything definitive myself so I trusted your comment. If you're also not sure, I'll just make the comment more vague :)

Reply:
If we are not sure, it might be nice to leave the original comment in its original location, and put a new comment in the other places we call `_convert_possible_tensor`